Make finding $OCAMLLIB work with Esy sandbox aliases

anmonteiro opened the pull request.

jaredly commented on src/analyze/BuildSystem.re:
```diff
@@ -223,8 +223,18 @@ let getStdlib = (base, buildSystem) => {
-    let%try_wrap esy_ocamllib = getLine("esy -q sh -- -c 'echo $OCAMLLIB'", ~pwd=base);
-    [esy_ocamllib]
+    let env = Unix.environment()->Array.to_list;
+    try {
```
I'd rather avoid such large `try` blocks. You're just dealing with the `Not_found` from `List.find`, right? Could you use `Belt.List.getBy` instead which returns an optional instead of throwing?

anmonteiro commented:
@jaredly thanks for the nice suggestions. I took the opportunity and cleaned up every `try` block that I had introduced. The code reads much nicer now.

jaredly commented:
perfect, thanks!
